velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -319,4 +319,19 @@ struct DayOfWeekFunction : public InitSessionTimezone<T>,
   }
 };
 
+template <typename T>
+struct DateDiffFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      int32_t& result,
+      const arg_type<Date>& endDate,
+      const arg_type<Date>& startDate) {
+    int64_t value = diffDate(DateTimeUnit::kDay, startDate, endDate);
```

Comment:
Can we not use endDate - startDate here?

Reply:
Yes, we can. Actually, no need to complicate it by calling diffDate(). Just updated.